File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    patterns = [
        [1,2,3,4,5],
        [2,1,2,3,2,4,2,5],
        [3,3,1,1,2,2,4,4,5,5]
    ]
    score = [0] * 3 # 각 학생 점수
    
    for idx, answer in enumerate(answers): #문제번호, 답
        for stu in range(3): #3명의 학생
            if answer == patterns[stu][idx%len(patterns[stu])]:
                score[stu]+=1
            # print(answer, stu, patterns[stu][idx%len(patterns[stu])])
    
    max_score = max(score)
    result = []
    
    for idx,value in enumerate(score):
        if value == max_score:
            result.append(idx+1)
    return sorted(result)

File: 프로그래머스/1/42840. 모의고사/모의고사.js
function solution(answers) {
    var answer = []; 
    const patterns = [
        [1,2,3,4,5],
        [2,1,2,3,2,4,2,5],
        [3,3,1,1,2,2,4,4,5,5]
    ]
    const score = [0,0,0]
    
    for(let i=0; i<3; i++){ // 학생 수
       let stu = patterns[i] // i번 학생 답
       for(let j=0; j<answers.length; j++){ // 정답
           if(answers[j] === stu[j%(stu.length)]) score[i]+=1
           // console.log(`학생 ${i+1}번, 답 : ${answers[j]} 학생 답 : ${stu[j%(stu.length)]}`)
       }
    }
    
    const max = Math.max(...score)
    
    for(let i=0; i<score.length; i++){
        if(score[i] === max) answer.push(i+1)
    }
    return answer;
}